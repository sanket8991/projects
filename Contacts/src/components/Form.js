import React, {useState} from 'react';
import {View, StyleSheet, Text, TextInput, Button} from 'react-native';

const Form= function({onSubmit, initialValues}) {
    const [name, setName] = useState(initialValues.name);
    const [contact,setContact] = useState(initialValues.contact);
    const [email, setEmail] = useState(initialValues.email);

    return(
    <View>
    <Text style = {styles.label}>Enter Name</Text>
    <TextInput 
        style = {styles.input} 
        value ={name}
        onChangeText = {(text) => setName(text)}
    />
    <Text style = {styles.label}>Enter Contact</Text>
    <TextInput 
        style = {styles.input} 
        value ={contact}
        keyboardType= "phone-pad"
        onChangeText = {(text) => setContact(text)}
    />

    <Text style = {styles.label}>Enter Email</Text>
    <TextInput 
        style = {styles.input} 
        value ={email}
        keyboardType= "email-address" 
        onChangeText = {(text) => setEmail(text)}
    />

    <Button 
        title='save'
        onPress={() => onSubmit(name,contact,email)}
    />

    </View>
    )
};

Form.defaultProps = {
    initialValues: {name:'', contact:'',email:''}
};



const styles = StyleSheet.create({
    label:{
        fontSize:20,
        marginBottom:10,
        marginLeft:5
    },

    input:{
        fontSize:18,
        borderWidth:1,
        borderColor: 'black',
        marginBottom:15,
        padding:5,
        margin:5

    }
});


export default Form;

