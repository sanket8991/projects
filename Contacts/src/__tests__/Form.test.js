import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Form from '../components/Form';
import {Alert} from 'react-native';

jest.spyOn(Alert,'alert');
const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
test('it renders the screen properly', () => {
    const {toJSON} = render(
    <Form onSubmit = {(name,contact,email,imagePath) => {
        if (!name || !contact){
            Alert.alert('Alert','the name and contact fields cannot be empty');

        }else if (!Number.isInteger(Number(contact)) || Number(contact) < 0){
            Alert.alert('Alert','the contact should contain only numbers')

        }else if (email !== "" && !email.toLowerCase().match(re)){
            Alert.alert('Alert','enter a valid email id');

        }else{
            dispatch(addContact({id:nanoid(), name:name, contact:String(Number(contact)),email:email,imagePath:imagePath}));
            Alert.alert('Alert','the contact was saved successfully.')
        }
    
    }}/>);
    expect(toJSON()).toMatchSnapshot();
});

test('it shows the alert for name and contact', () => {
    const {getByTestId} = render(
    <Form onSubmit = {(name,contact,email,imagePath) => {
        if (!name || !contact){
            Alert.alert('Alert','the name and contact fields cannot be empty');

        }else if (!Number.isInteger(Number(contact)) || Number(contact) < 0){
            Alert.alert('Alert','the contact should contain only numbers')

        }else if (email !== "" && !email.toLowerCase().match(re)){
            Alert.alert('Alert','enter a valid email id');

        }else{
            Alert.alert('Alert','the contact was saved successfully.')
        }
    
    }}/>);
    // for name or contact field being left empty
    const inputName = getByTestId('nameInput');
    const inputContact = getByTestId('contactInput');
    fireEvent.changeText(inputName, 'cat');
    fireEvent.press(getByTestId('submitButton'));
    expect(inputName.props.value).toBe('cat')
    expect(Alert.alert).toHaveBeenCalledWith('Alert','the name and contact fields cannot be empty')


});

test('it shows the alert invalid contact input', () => {
    const {getByTestId} = render(
    <Form onSubmit = {(name,contact,email,imagePath) => {
        if (!name || !contact){
            Alert.alert('Alert','the name and contact fields cannot be empty');

        }else if (!Number.isInteger(Number(contact)) || Number(contact) < 0){
            Alert.alert('Alert','the contact should contain only numbers')

        }else if (email !== "" && !email.toLowerCase().match(re)){
            Alert.alert('Alert','enter a valid email id');

        }else{
            Alert.alert('Alert','the contact was saved successfully.')
        }
    
    }}/>);
    // for name or contact field being left empty
    const inputName = getByTestId('nameInput');
    const inputContact = getByTestId('contactInput');
    fireEvent.changeText(inputName, 'cat');
    fireEvent.changeText(inputContact, '123.123.123')
    fireEvent.press(getByTestId('submitButton'));
    expect(inputName.props.value).toBe('cat')
    expect(Alert.alert).toHaveBeenCalledWith('Alert','the contact should contain only numbers')


});


test('it shows the alert invalid email input', () => {
    const {getByTestId} = render(
    <Form onSubmit = {(name,contact,email,imagePath) => {
        if (!name || !contact){
            Alert.alert('Alert','the name and contact fields cannot be empty');

        }else if (!Number.isInteger(Number(contact)) || Number(contact) < 0){
            Alert.alert('Alert','the contact should contain only numbers')

        }else if (email !== "" && !email.toLowerCase().match(re)){
            Alert.alert('Alert','enter a valid email id');

        }else{
            Alert.alert('Alert','the contact was saved successfully.')
        }
    
    }}/>);
    // for name or contact field being left empty
    const inputName = getByTestId('nameInput');
    const inputContact = getByTestId('contactInput');
    const inputEmail = getByTestId('emailInput');
    fireEvent.changeText(inputName, 'cat');
    fireEvent.changeText(inputContact, '123456');
    fireEvent.changeText(getByTestId("emailInput"), 'someemail')
    fireEvent.press(getByTestId('submitButton'));
    expect(inputName.props.value).toBe('cat')
    expect(Alert.alert).toHaveBeenCalledWith('Alert','enter a valid email id')


});
