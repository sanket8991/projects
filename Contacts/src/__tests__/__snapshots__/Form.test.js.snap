// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the screen properly 1`] = `
<RCTScrollView>
  <View>
    <View>
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 250,
            "width": 250,
          }
        }
      />
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Add your photo
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "marginBottom": 10,
            "marginLeft": 5,
          }
        }
      >
        Enter Name
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 1,
            "fontSize": 18,
            "margin": 5,
            "marginBottom": 15,
            "padding": 5,
          }
        }
        testID="nameInput"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 20,
            "marginBottom": 10,
            "marginLeft": 5,
          }
        }
      >
        Enter Contact
      </Text>
      <TextInput
        keyboardType="phone-pad"
        maxLength={13}
        onChangeText={[Function]}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 1,
            "fontSize": 18,
            "margin": 5,
            "marginBottom": 15,
            "padding": 5,
          }
        }
        testID="contactInput"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 20,
            "marginBottom": 10,
            "marginLeft": 5,
          }
        }
      >
        Enter Email
      </Text>
      <TextInput
        keyboardType="email-address"
        onChangeText={[Function]}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 1,
            "fontSize": 18,
            "margin": 5,
            "marginBottom": 15,
            "padding": 5,
          }
        }
        testID="emailInput"
        value=""
      />
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="submitButton"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            save
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
